### PR TITLE
doc: Update data for v3.12.2 release

### LINKED ISSUE #331: Release v3.12.2
#### Overview

This is a checklist for releases. This is filled in by both the releaser and the reviewer where necessary.

#### Update Documentation

- [x] Create a new release in the [`whats-new` section](doc/content/whats-new) and copy the release CHANGELOG from [TheThingsIndustries/lorawan-stack](https://github.com/TheThingsIndustries/lorawan-stack). The title is the release version and the date is the release date.
- [x] Update the [documentation version](https://github.com/TheThingsIndustries/lorawan-stack-docs/blob/master/doc/config/_default/config.toml#L28) to match the current minor, if necessary (`v3.${minor}`).
- [x] To generate documentation, create a clone of [TheThingsIndustries/lorawan-stack](https://github.com/TheThingsIndustries/lorawan-stack), and **checkout the git tag of the release**.
- [x] To generate API documentation, run the following from within the clone of [TheThingsIndustries/lorawan-stack](https://github.com/TheThingsIndustries/lorawan-stack): 

```bash
$ tools/bin/mage ttiProto:hugoData
```

- [x] Copy generated files to `lorawan-stack-docs` with the following command:

```bash
$ cp api/ttn.lorawan.v3/*.yml ../lorawan-stack-docs/doc/data/api/ttn.lorawan.v3/
$ cp api/tti.lorawan.v3/*.yml ../lorawan-stack-docs/doc/data/api/tti.lorawan.v3/
```

> NOTE: Replace `../lorawan-stack-docs` if your `lorawan-stack-docs` folder is in a different location.

- [x] To generate and export CLI documentation from within the clone of [TheThingsIndustries/lorawan-stack](https://github.com/TheThingsIndustries/lorawan-stack), first run the following command to build the CLI:

```bash
$ go build -tags tti ./cmd/tti-lw-cli
```

- [x] Then export the CLI documentation using the following command:

```bash
$ HOME='$HOME' ./tti-lw-cli gen-yaml-doc -o /path/to/lorawan-stack-docs/doc/data/
```

> NOTE: Replace `../lorawan-stack-docs` if your `lorawan-stack-docs` folder is in a different location.

#### Check (for reviewers)

- [ ] A new section has been created in [`whats-new`](doc/content/whats-new) with the corresponding CHANGELOG from [TheThingsIndustries/lorawan-stack](https://github.com/TheThingsIndustries/lorawan-stack). The title is the release version and the date is the release date.
- [ ] The documentation version is up to date.
- [ ] The TTI and TTN API documentation has been generated and updated in [doc/data/api](https://github.com/TheThingsIndustries/lorawan-stack-docs/blob/master/doc/data/api). This includes the following files:

```
doc/data/api/tti.lorawan.v3/messages.yml
doc/data/api/tti.lorawan.v3/services.yml
doc/data/api/tti.lorawan.v3/enums.yml
doc/data/api/ttn.lorawan.v3/messages.yml
doc/data/api/ttn.lorawan.v3/services.yml
doc/data/api/ttn.lorawan.v3/enums.yml
```

- [ ] The TTI CLI documentation has been generated and updated in [doc/data](https://github.com/TheThingsIndustries/lorawan-stack-docs/blob/master/doc/data). This includes the following file:

```
doc/data/ttn-lw-cli.yml
```

- [ ] All generated documentation matches the version that is being released.

### PR DESCRIPTION
**Pull request by nejraselimovic**
#### Summary
Closes #331 

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Run Locally: Verified that the docs build using `make server`, posted screenshots, verified external links. Test with `HUGO_PARAMS_SEARCH_ENABLED=true` if style changes will affect the search bar.
- [ ] New Features Marked: Documentation for new features is marked using the `new-in-version` shortcode, according to the guidelines in [CONTRIBUTING](CONTRIBUTING.md).
- [ ] Style Guidelines: Documentation obeys style guidelines in [CONTRIBUTING](CONTRIBUTING.md).
- [ ] Commits: Commit messages follow guidelines in [CONTRIBUTING](CONTRIBUTING.md), there are no fixup commits left.

### CODE REVIEW

File: doc/content/whats-new/3.12.2.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+### Fixed
+
+- Occasional crashes in the ratelimit middleware.
+- Handling of zero EUI CUPS update-info requests.
+- Backend validation messages for some forms.
+- Gateway downlink message previews not displaying correctly in the event view of the Console.
+- Importing end devices from the Console would occasionally ignore some device MAC settings fields.
+
+### Security
```
Comment (htdvisser) [inline doc/content/whats-new/3.12.2.md:29]: Shouldn't we remove empty sections?

File: doc/data/ttn-lw-cli.yml [hunk 2]
```diff
@@ -22,7 +22,7 @@ ttn-lw-cli:
     type: stringSlice
     shorthand: c
     usage: Location of the config files
-    default_value: '[.ttn-lw-cli.yml,$HOME/.ttn-lw-cli.yml,$HOME/Library/Application Support/.ttn-lw-cli.yml]'
+    default_value: '[.ttn-lw-cli.yml,$HOME/.ttn-lw-cli.yml,$HOME/.config/.ttn-lw-cli.yml]'
```
Comment (htdvisser) [inline doc/data/ttn-lw-cli.yml:25]: This is sort of expected when generating the docs on macOS vs Linux. I don't think we can avoid this.

**Review by htdvisser — APPROVED** [review]
Looks good to me 👍

File: doc/content/whats-new/3.12.2.md (hunk 1 above)
Comment (nejraselimovic) [reply doc/content/whats-new/3.12.2.md:29]: These aren't removed for previous releases, so I will leave this for now. If we decide to remove those, I can easily remove it in another PR

**Pull request merged**